### ISSUE TITLE
No proper documentation on how to do PKCE flow with Azure AD

### ISSUE THREAD

**Issue by SuperMohit** [nextauthjs/next-auth#7114]
### What is the improvement or update you wish to see?

I am trying to get up and running with PKCE flow using Azure AD. But there is no clear documentation whatsoever.
The OAuth flow works with the default documentation. But fails when I try to do pkce flow. It will be helpful to have this documented somewhere.

### Is there any context that might help us understand?

 This is the config I gave to run the pkce flow. Is this correct? I am using nuxt3 version of this so config looks like this:
```
 AzureAD.default({
        checks: ['state', 'pkce'],
        clientId: '<redacted>',
        tenantId: '<redacted>',
      }),
```

When I try this, I get the following error:
```
    name: 'OAuthCallbackError',
    code: undefined
  },
  providerId: 'azure-ad',
  message: 'client_secret_basic client authentication method requires a client_secret'
```

**Comment by SuperMohit** [comment]
@balazsorban44 I am trying to make PKCE work from localhost using Azure AD. Can you please help or point me to someone who can help?

**Comment by balazsorban44** [comment]
As the error suggests, `client_secret` is missing. 

Auth.js is usually used in scenarios where you have a server. From what you are telling, it is most likely that you are doing this too (You say it's a Nuxt app), so the easiest way is to add the required `clientSecret` as well.

Note, although technically client secret is not required with PKCE, it is good to define it, when you have a backend to work with and you don't work on a public client like a mobile app. In this case, Nuxt is mentioned which does have a backend. I cannot say much more as you did not attach a reproduction/much code.

That said, you _can_ configure the OAuth client to not require `client_secret` - which I saw we forgot to document - so I just pushed a small update. See https://authjs.dev/reference/core/providers#client

These options require knowledge of the OAuth spec, so proceed with caution.
